### PR TITLE
Report non-boolean outputs UNDETERMINED and UNREACHABLE

### LINKED ISSUE #27: Provide non-boolean options as result
The extension currently provides only success or failure as the output options, when there is a possibility for more than that to be returned as the output from Kani i.e (undetermined | unreachable). The user needs to be alerted to these outputs as well, when they do happen and an option for non failure UI should be designed and provided to the user.

Issues arises from this part of the code - https://github.com/model-checking/kani-vscode-extension/blob/3c805273e201505eb576235505252f5cb02fcb86/src/model/kaniOutputParser.ts#L98

### PR DESCRIPTION
**Pull request by qinheping**
### Description of changes: 

The extension currently provides only success or failure as the output options. With this PR, it will also report UNDETERMINED and UNREACHABLE, which are also possible status returned from Kani.

Here is an example when assertions are `UNDETERMINED`.

Screenshot 2023-05-18 at 1 34 59 PM

### Resolved issues:

Resolves #27 

### Related RFC:

Optional #ISSUE-NUMBER.

### Call-outs:

When the verification successes, we don't report result of `cover` checks. 
@adpaco-aws  should we say a verification is failed if some `cover` checks are unsatisfiable?

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: src/model/kaniOutputParser.ts [hunk 1]
```diff
@@ -101,9 +101,20 @@ function parseChecksArray(checksArray: Array<string>): KaniResponse {
...
+			checkInstanceObject.status == 'UNREACHABLE' ||
+			checkInstanceObject.status == 'SATISFIED' ||
+			checkInstanceObject.status == 'UNSATISFIABLE'
+		) {
 			failureResponseMessage += checkInstanceObject.createFailureMessage() + '\n';
 			failureDisplayMessage = checkInstanceObject.createDisplayMessage() + '\n';
+		} else if (checkInstanceObject.status != 'SUCCESS') {
+			failureResponseMessage +=
+				checkInstanceObject.createFailureMessage() +
+				'WARNING: unknown status returned from Kani.\n\n';
```
Comment (karkhaz) [inline src/model/kaniOutputParser.ts:116]: Suggest that you include the value of `checkInstanceObject.status` in the error message?

File: src/model/kaniOutputParser.ts [hunk 2]
```diff
@@ -101,9 +101,20 @@ function parseChecksArray(checksArray: Array<string>): KaniResponse {
...
 		const checkInstance: string[] = checksArray[i].split('\n');
 		const checkInstanceObject: CheckInstance = convertChecktoObject(checkInstance);
-		if (checkInstanceObject.status == 'FAILURE') {
+		if (
+			checkInstanceObject.status == 'FAILURE' ||
+			checkInstanceObject.status == 'UNDETERMINED' ||
+			checkInstanceObject.status == 'UNREACHABLE' ||
+			checkInstanceObject.status == 'SATISFIED' ||
+			checkInstanceObject.status == 'UNSATISFIABLE'
+		) {
```
Comment (karkhaz) [inline src/model/kaniOutputParser.ts:110]: nit, IMHO this is neater:
```suggestion
                const error_statuses = [                          
                    'FAILURE', 'UNDETERMINED', 'UNREACHABLE', 'SATISFIED', 'UNSATISFIABLE'                                 
                ];
		if (
		    error_statuses.includes(checkInstanceObject.status)
		) {
```

File: src/model/kaniOutputParser.ts (hunk 1 above)
Comment (qinheping) [reply src/model/kaniOutputParser.ts:116]: Yes, the value of `checkInstanceObject.status` is already included in `checkInstanceObject.createFailureMessage()`.

File: src/model/kaniOutputParser.ts (hunk 2 above)
Comment (qinheping) [reply src/model/kaniOutputParser.ts]: Fixed. Thank you!

**Review by karkhaz — APPROVED** [review]
thank you!

File: src/model/kaniOutputParser.ts [hunk 3]
```diff
@@ -97,13 +97,19 @@ function cleanChecksArray(checksArray: Array<string>): Array<string> {
 function parseChecksArray(checksArray: Array<string>): KaniResponse {
 	let failureResponseMessage = ``;
 	let failureDisplayMessage = ``;
+	const error_statuses = ['FAILURE', 'UNDETERMINED', 'UNREACHABLE', 'SATISFIED', 'UNSATISFIABLE'];
```
[the remark this answers is not in the document]
Comment (qinheping) [reply src/model/kaniOutputParser.ts]: Fixed.
SATISFIED is a tricky one that, although it is a success status, users may want to see the trace for it. I agree to move it out from the failure statues now. How to handle `cover` checks and generate concrete examples for them is another issue beyond the scope of this PR.

[the remark this answers is not in the document]
Comment (jaisnan) [reply src/model/kaniOutputParser.ts]: Thanks!

**Review by jaisnan — APPROVED** [review]
LGTM!

**Pull request merged**